File: backend/cs/SocialNetApp/Bff/API/Dtos/UserDto.cs
using System.ComponentModel.DataAnnotations;

namespace Bff.API.Dtos
{
    public readonly record struct UserDto
    {
        [Required]
        public uint Id { get; init; }
        [Required]
        public string Name { get; init; }
        [Required]
        public string Surname { get; init; }
        [Required]
        public string Password { get; init; }
        public byte? Age { get; init; }
        public string Sex { get; init; }
        public string City { get; init; }
        public string Info { get; init; }
    }
}

File: backend/go/socialnetapp/bff/internal/handler/dto/userdto.go
package dto

// UserDto represents dto for a user
type UserDto struct {
	ID      uint    `json:"id"`
	Name    string  `json:"name"`
	Surname string  `json:"surname"`
	Age     *uint8  `json:"age"`
	Sex     *string `json:"sex"`
	City    *string `json:"city"`
	Info    *string `json:"info"`
}
